tasks/collect-marketplacesvm-params/tests/test-collect-marketplacesvm-params-pre-push.yaml

```diff
@@ -0,0 +1,79 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-collect-marketplacesvm-params-pre-push
+spec:
+  description: |
+    Run the collect-marketplacesvm-params task with the secret required and verify that
+    it will return the secret string.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-values
+            image: quay.io/konflux-ci/release-service-utils:6556e8a6b031c1aad4f0472703fd121a6e1cd45d
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.data.path)/data.json << EOF
```

Comment:
@mmalina how is checkton not complaining for this?

Reply:
Good question... @chmeliik any idea? In the workflow log I can see:
```
Files to check:
...
C079	tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml	tasks/collect-marketplacesvm-params/tests/test-collect-marketplacesvm-params-pre-push.yaml
```

Reply:
What is the issue here?

Reply:
We expect checkton to report that you don't have `"` around `$(workspaces.data.path)/data.json`. We have been fixing checkton in all of our tasks and pretty much every task complained about it. But checkton is passing here, which we find strange

Reply:
Gotcha! Should I patch it or leave this for you to investigate more on checkton?

Reply:
Maybe it is due to the use of `sh` instead of `bash`? checkton passes so it is okay for you to leave it. We don't want to start mandating things in addition to what checkton tells you to do

Reply:
> Maybe it is due to the use of sh instead of bash?

I don't think so. If I copy and paste the first few lines at https://www.shellcheck.net it will complain.

Reply:
> Good question... @chmeliik any idea?

Yeah, what happened here is this:

```
  C079	tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml	tasks/collect-marketplacesvm-params/tests/test-collect-marketplacesvm-params-pre-push.yaml
  R082	tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml	tasks/collect-marketplacesvm-params/tests/test-collect-marketplacesvm-params.yaml
```

The base branch had a `tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml` file. This PR has two files, one is an 79% Copy and the other is an 82% Rename of the file from the base branch.

Checkton compares shellcheck warnings with those that already exist in the base branch and doesn't report them. This include handling for copies and renames (warnings in a copied/renamed file won't get reported if the base file already had them)

You can tweak the behavior with some params if you want https://github.com/chmeliik/checkton?tab=readme-ov-file#find-renames

Reply:
Okay, @JAVGan can you add the quotes

Reply:
Sure, adding them

Reply:
I've added them, PTAL again.

I've also rebased the PR

Reply:
@chmeliik thanks for the explanation. But doesn't that mean that this should have been reported when the file in the base branch was added? See here: https://github.com/konflux-ci/release-service-catalog/blob/a7ee201db4bee2044c8b1f0e325302ad8df1e896/tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml#L27

It was added here: https://github.com/konflux-ci/release-service-catalog/pull/719

That PR has the line `cat > $(workspaces.data.path)/data.json << EOF`.

From the checkton run there:
```
C069	tasks/collect-registry-token-secret/tests/test-collect-registry-token-secret.yaml	tasks/collect-marketplacesvm-secret/tests/test-collect-marketplacesvm-secret.yaml
```

So once again, it treated it as a copy of existing `tasks/collect-registry-token-secret/tests/test-collect-registry-token-secret.yaml`. And that one had the same line. It was added in September: https://github.com/konflux-ci/release-service-catalog/pull/523 Checkton succeded there, but there are no logs - probably too old.

So that's interesting.

This wouldn't be an issue if everything our repo was fixed for checkton. We're working on that here and it's almost done: https://issues.redhat.com/browse/RELEASE-1032

@johnbieren maybe we should add a jira to run checkton on everything and see if there are any leftovers - there likely will be because of this.

Reply:
Yeah, if all the test files are roughly copies of each other, the default behavior will end up hiding a lot of warnings. You may want to disable the handling for copies or to disable differential mode altogether once you've fixed all the warnings

Reply:
If we didn't work on fixing everything, I would say we should set find-renames and find-copies to false. (BTW, @johnbieren you were worried that if you moved all the files before we finish fixing this, you would have all the checkton warnings - as we can see here, you wouldn't ;) )

> You may want to disable the handling for copies or to disable differential mode altogether once you've fixed all the warnings

Good point. I am not sure we should do it. Maybe... That would be similar to how tests in konflux-release-data are run. On the other hand in this repo we only run tekton task unit tests for tasks that are being changed. So maybe it makes sense to leave checkton similar to that.

Reply:
> @johnbieren maybe we should add a jira to run checkton on everything once all the other stories are done and see if there are any leftovers - there likely will be because of this.

+1

I am fine with us doing a one off to fix lingering things. None of the checkton issues were that hard to fix, and there are only a few unique things to do so we can just copy what we did in other tasks. In other words, that ticket should be straight forward.

Going forward, I'd like checkton to run on any file in the diff. So if you update line 8 of file x/y/z, z should fail checkton even if the violating line is in line 20 of file z or if z is similar to file x/a/b. Is that a possible config setting @chmeliik ?

Reply:
If I understand correctly, you want to disable finding renames and copies https://github.com/chmeliik/checkton?tab=readme-ov-file#find-renames